### PR TITLE
[read-fonts] initial codegen for parsing sbix table

### PR DESCRIPTION
**Pull request by dfrg**
Starting work on bitmap support and figured I'd tackle the low hanging fruit first. This includes basic codegen for parsing sbix along with a tiny helper method to load `GlyphData` for a given glyph id.

Working towards #593

### CODE REVIEW

File: resources/codegen_inputs/sbix.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#[read_args(num_glyphs: u16)]
+#[tag = "sbix"]
+table Sbix {
+    /// Table version number — set to 1.
+    #[compile(1)]
+    version: u16,
+    /// Bit 0: Set to 1.
+    /// Bit 1: Draw outlines.
+    /// Bits 2 to 15: reserved (set to 0).
+    flags: u16,
```
Comment (cmyr) [inline resources/codegen_inputs/sbix.rs:13]: we might want to define a flags type for this?

**Review by cmyr — APPROVED** [review]
looks good, I think matching the glyf/loca API is the right idea (not that there is another clear alternative, so 🤷)

File: resources/codegen_inputs/sbix.rs (hunk 1 above)
Comment (dfrg) [reply resources/codegen_inputs/sbix.rs:13]: Good call. The unnamed bit 0 that must always be set to 1 seems problematic for the codegen flags type. Wdyt about wrapping in a record instead and implementing default and appropriate methods for the “render outlines bit” instead?

Comment (cmyr) [reply resources/codegen_inputs/sbix.rs:13]: I think an easier option would be to use flags, but use the `compile_with` annotation to provide a fn that ensures that bit one is always set when writing.

Comment (dfrg) [reply resources/codegen_inputs/sbix.rs:13]: Ah, will do!

Comment (dfrg) [reply resources/codegen_inputs/sbix.rs:13]: Since I'm adding `compile_with`, went ahead and extended this for write support. Should be good to go.

**Pull request merged**